### PR TITLE
IRGen: Add code to support building fragile resilient protocol witnesses

### PR DESCRIPTION
**Pull request by aschwaighofer**
We don't currently support building resilient relative protocol witness tables. One might want to build with relative witness tables but not need resilient protocols. Allow for that scenario.

Add a test configuration to test library-evolution + fragile resilient protocols + relative protocol witness tables.

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci test

File: lib/IRGen/GenProto.cpp [hunk 1]
```diff
@@ -2528,6 +2532,16 @@ void IRGenModule::emitSILWitnessTable(SILWitnessTable *wt) {
   bool isResilient = isResilientConformance(conf);
   bool useRelativeProtocolWitnessTable =
     IRGen.Opts.UseRelativeProtocolWitnessTables;
+  if (useRelativeProtocolWitnessTable &&
```
Comment (aschwaighofer) [inline lib/IRGen/GenProto.cpp:2535]: @slavapestov I am trying to test for variadic generics with conditional conformances here. Probably, doing it all wrong?

Comment (aschwaighofer) [reply lib/IRGen/GenProto.cpp:2535]: Well, for one a `dump()` isn't going to fly in an no-asserts build ...

**Comment by aschwaighofer** [comment]
@swift-ci test

File: lib/IRGen/GenProto.cpp [hunk 2]
```diff
@@ -2528,6 +2532,18 @@ void IRGenModule::emitSILWitnessTable(SILWitnessTable *wt) {
...
+      !conf->getConditionalRequirements().empty()) {
+    auto nominal = conf->getType()->getAnyNominal();
+    auto sig = nominal->getGenericSignatureOfContext();
```
Comment (slavapestov) [inline lib/IRGen/GenProto.cpp:2538]: This should work too: `sig = conf->getGenericSignature();`

File: lib/IRGen/GenProto.cpp (hunk 1 above)
Comment (slavapestov) [reply lib/IRGen/GenProto.cpp:2535]: It looks fine :)

**Review by slavapestov — APPROVED** [review]

**Comment by aschwaighofer** [comment]
@swift-ci test

**Review by kalebers — APPROVED** [review]

**Pull request merged**